src/com/etendoerp/dependencymanager/process/AddDependency.java

```diff
@@ -67,56 +67,67 @@ protected JSONObject execute(Map<String, Object> parameters, String data) {
-        packageDependencyCriteria.add(Restrictions.ne(PackageDependency.PROPERTY_ARTIFACT, PackageUtil.ETENDO_CORE));
-        List<PackageDependency> dependencyList = packageDependencyCriteria.list();
-
-        boolean needFlush = false;
-        for (PackageDependency packageDependency : dependencyList) {
-          boolean isExternalDependency = packageDependency.isExternalDependency().booleanValue();
-          if (DependencyUtil.existsDependency(packageDependency.getGroup(), packageDependency.getArtifact(),
-              packageDependency.getVersion(),
-              isExternalDependency)) {
-            log.debug("Dependency already exists: %s:%s:%s", packageDependency.getGroup(),
-                packageDependency.getArtifact(), packageDependency.getVersion());
-            continue;
-          }
-          Dependency dependency = new Dependency();
-          dependency.setVersion(packageDependency.getVersion());
-          dependency.setGroup(packageDependency.getGroup());
-          dependency.setArtifact(packageDependency.getArtifact());
-          dependency.setInstallationStatus(DependencyUtil.STATUS_PENDING);
-          String versionStatus;
-          if (isExternalDependency) {
-            dependency.setFormat(DependencyUtil.FORMAT_JAR);
-            dependency.setExternalDependency(true);
-            versionStatus = DependencyUtil.UNTRACKED_STATUS;
-          } else {
-            dependency.setFormat(DependencyUtil.FORMAT_SOURCE);
-            PackageVersion latestPackageVersion = PackageUtil.getLastPackageVersion(packageDependency.getDependencyVersion().getPackage());
-            versionStatus = InstallDependency.determineVersionStatus(packageDependency.getVersion(),
-                latestPackageVersion.getVersion());
-          }
-          dependency.setVersionStatus(versionStatus);
-          needFlush = true;
-          OBDal.getInstance().save(dependency);
-          log.debug("New dependency added: %s", dependency.getIdentifier());
-        }
+        String fromCore = packageVersion.getFromCore();
+        String latestCore = packageVersion.getLatestCore();
 
-        needFlush |= addVersionToInstall(packageVersion, dependencyList);
-        if (needFlush) {
-          OBDal.getInstance().flush();
+        if (fromCore == null || latestCore == null) {
+          successMessage = String.format(
+              OBMessageUtils.messageBD("ETDEP_Invalid_Core_Versions"), packageVersion.getPackage().getIdentifier(),
+              packageVersion.getVersion());
+          log.warn("Invalid core versions for package %s in version %s", packageVersion.getPackage().getIdentifier(),
+              packageVersion.getIdentifier());
+        } else {
           successMessage = String.format(
-              OBMessageUtils.messageBD("ETDEP_Added_Dependencies"), packageVersion.getPackage().getIdentifier(),
+              OBMessageUtils.messageBD("ETDEP_Not_Dependencies"), packageVersion.getPackage().getIdentifier(),
```

Comment:
This message doesn't exists, you must be more careful when resolve conflicts, because this was fixed in dev after your brach was created.